sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ResourceProviderDataTests.cs

```diff
@@ -83,5 +83,54 @@ public void ValidateCapitalizationIsIgnored()
             Assert.AreEqual(File.ReadAllText(TestData.GetLocation("ResourceProviderData/ResourceProviderData-TwoSpaces.json")).TrimEnd(), stjResult);
 #endif
         }
+
+        [Test]
+        public void ProxySerialization()
```

Comment:
Would it be possible to add a test that illustrates adding multiple proxies -- one for the top-level model, and at least one for a model corresponding to a nested model type?

That is to say -- if model A looks like

class A
{
  B PropertyB { get; set; }
}

class B
{
  C PropertyC { get; set; }
}

what would it look like to do:

options = new();
options.AddProxy(typeof(A), new ProxyForA());
options.AddProxy(typeof(C), new ProxyForC());

?